-= BEGIN original header =-
Original patch from gentoo: gentoo/src/patchsets/gcc/4.3.1/uclibc/90_all_304-index_macro.patch
-= END original header =-

diff -durN gcc-4.4.5.orig/libstdc++-v3/include/ext/rope gcc-4.4.5/libstdc++-v3/include/ext/rope
--- gcc-4.4.5.orig/libstdc++-v3/include/ext/rope	2009-04-10 01:23:07.000000000 +0200
+++ gcc-4.4.5/libstdc++-v3/include/ext/rope	2010-10-09 23:12:51.000000000 +0200
@@ -54,6 +54,9 @@
 #include <bits/gthr.h>
 #include <tr1/functional>
 
+/* cope w/ index defined as macro, SuSv3 proposal */
+#undef index
+
 # ifdef __GC
 #   define __GC_CONST const
 # else
diff -durN gcc-4.4.5.orig/libstdc++-v3/include/ext/ropeimpl.h gcc-4.4.5/libstdc++-v3/include/ext/ropeimpl.h
--- gcc-4.4.5.orig/libstdc++-v3/include/ext/ropeimpl.h	2009-04-10 01:23:07.000000000 +0200
+++ gcc-4.4.5/libstdc++-v3/include/ext/ropeimpl.h	2010-10-09 23:12:51.000000000 +0200
@@ -49,6 +49,9 @@
 #include <ext/memory> // For uninitialized_copy_n
 #include <ext/numeric> // For power
 
+/* cope w/ index defined as macro, SuSv3 proposal */
+#undef index
+
 _GLIBCXX_BEGIN_NAMESPACE(__gnu_cxx)
 
   using std::size_t;
